eclipse-sw360/sw360
# feat(rest):update response api get attachment by component

[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)


Issue: #1977


### How To Test?

- Run command line: 
```
$ curl 'https://sw360.org/api/components/17653524/attachments' -i -X GET \
    -H 'Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOlsic3czNjAtUkVTVC1BUEkiXSwidXNlcl9uYW1lIjoiYWRtaW5Ac3czNjAub3JnIiwic2NvcGUiOlsiYWxsIl0sImV4cCI6MTY4NTYxOTg0MiwiYXV0aG9yaXRpZXMiOlsiUkVBRCIsIldSSVRFIl0sImp0aSI6IjBISkVRRFlrbS1WSVBCMWJPekZ3YkszLWtBZyIsImNsaWVudF9pZCI6InRydXN0ZWQtc3czNjAtY2xpZW50In0.Glx7OHqf4MC2mPOBgW0YRyiaR6P3plIdhGeMsoM_e1Uwry5oheVrDFSdmfyw-LOywODcvsJzlTF1RlHhjYgsToVMEnv0wFg5yV5OSObHVbML-fksMbIRYw95hJT9IhNn4pfMNOTU2loVS8f7GW2wZhC6sjkpBrJFG9IcdwMtz83RR5NZgvgI6t8MZght8GLKKRwk9FCc4asrxhQw134Mb0-4VL8vms7u2aaIMcSi0ytL2qHii4upDubjIqlKRrAZt7ms3T94pESLJZA2hbxFYwoNHpoaWUCUrrd5qvksG2SCBAMu34cKcw36TkuYiX96XxTUlWyXkjRHFPSpV090jA' \
    -H 'Accept: application/hal+json'
```
- Expected Output:
```
{
  "_embedded" : {
    "sw360:attachmentDTOes" : [ {
      "attachmentContentId" : "",
      "filename" : "spring-core-4.3.4.RELEASE.jar",
      "sha1" : "da373e491d3863477568896089ee9457bc316783",
      "attachmentType" : "BINARY_SELF",
      "createdBy" : "admin@sw360.org",
      "createdTeam" : "Clearing Team 1",
      "createdComment" : "please check asap",
      "createdOn" : "2016-12-18",
      "checkedTeam" : "Clearing Team 2",
      "checkedComment" : "everything looks good",
      "checkedOn" : "2016-12-18",
      "checkStatus" : "ACCEPTED",
      "usageAttachment" : {
        "visible" : 0,
        "restricted" : 0
      }
    } ]
  },
  "_links" : {
    "curies" : [ {
      "href" : "https://sw360.org/docs/{rel}.html",
      "name" : "sw360",
      "templated" : true
    } ]
  }
}
```
